src/components/ui/coaching-sessions/tiptap-editor.tsx

```diff
@@ -1,266 +1,94 @@
-          <Button
-            variant="ghost"
-            onClick={() => editor.chain().focus().toggleUnderline().run()}
-            className={`p-2 mr-0.5 rounded ${
-              editor.isActive("underline") ? "button-active" : ""
-            }`}
-            title="Underline (Ctrl+U)"
-          >
-            <UnderlineIcon className="h-4 w-4" />
-          </Button>
-
-          {/* Strikethrough Button */}
-          <Button
-            variant="ghost"
-            onClick={() => editor.chain().focus().toggleStrike().run()}
-            className={`p-2 mr-0.5 rounded ${
-              editor.isActive("strike") ? "button-active" : ""
-            }
-            `}
-            title="Strike Through"
-          >
-            <Strikethrough className="h-4 w-4" />
-          </Button>
-
-          {/* Highlight Button */}
-          <Button
-            variant="ghost"
-            onClick={() => editor.chain().focus().toggleHighlight().run()}
-            className={`p-2 mr-0.5 rounded ${
-              editor.isActive("highlight") ? "button-active" : ""
-            }`}
-            title="Highlight Text"
-          >
-            <Highlighter className="h-4 w-4" />
-          </Button>
-
-          {/* Heading 1 */}
-          <Button
-            variant="ghost"
-            onClick={() =>
-              editor.chain().focus().toggleHeading({ level: 1 }).run()
-            }
-            className={`p-2 mr-0.5 rounded ${
-              editor.isActive("heading", { level: 1 }) ? "button-active" : ""
-            }`}
-            title="Heading1"
-          >
-            <Heading1 className="h-4 w-4" />
-          </Button>
-
-          {/* Heading */}
-          <Button
-            variant="ghost"
-            onClick={() =>
-              editor.chain().focus().toggleHeading({ level: 2 }).run()
-            }
-            className={`p-2 mr-0.5 rounded ${
-              editor.isActive("heading", { level: 2 }) ? "button-active" : ""
-            }`}
-            title="Heading2"
-          >
-            <Heading2 className="h-4 w-4" />
-          </Button>
-
-          {/* Heading 3 */}
-          <Button
-            variant="ghost"
-            onClick={() =>
-              editor.chain().focus().toggleHeading({ level: 3 }).run()
-            }
-            className={`p-2 mr-0.5 rounded ${
-              editor.isActive("heading", { level: 3 }) ? "button-active" : ""
-            }`}
-            title="Heading3"
-          >
-            <Heading3 className="h-4 w-4" />
-          </Button>
+    // Only initialize the provider when JWT is available and there are no errors
+    if (!isLoading && !isError && jwt) {
+      const newProvider = new TiptapCollabProvider({
+        name: jwt.sub,
+        appId: tiptapAppId,
+        token: jwt.token,
+        document: doc,
+        user: userSession.display_name,
+        connect: true,
+        broadcast: true,
+      });
+
+      setProvider(newProvider);
```

Comment:
I'm still setting getting branches running up so I could just be totally wrong, but we may not need the stateful reference to the provider. Will test this theory if when I get it up and running and confirm or correct myself.

Reply:
Yeah you're probably totally right. This is one of those AI decisions that I wasn't sure about but went with, blindly, because it seemed to work. I'd be very interested to get your thoughts on how this should be set up.

Reply:
And some more [here](https://github.com/ueberdosis/hocuspocus/blob/main/docs/provider/examples.md) too (examples).